src/parser/core/modules/Procs.tsx

```diff
@@ -217,7 +215,35 @@ export abstract class Procs extends Analyser {
 	protected getDropCountForStatus(status: number| ProcGroup): number {
 		const procGroup = this.getTrackedGroupByStatus(status)
 		if (procGroup == null) { return 0 }
-		return this.getRemovalCountForStatus(status) - this.getUsageCountForStatus(status)
+		const stacksPerWindow = procGroup.procStatus.stacksApplied ?? 1
+
+		return this.getHistoryForStatus(status)
+			.filter(window => window.overwritten === false && window.consumingEvents.length < stacksPerWindow)
+			.reduce((dropped, window) => {
+				if (this.considerDroppedProcs(window)) {
+					dropped += Math.max(0, stacksPerWindow - window.consumingEvents.length)
+				}
+				return dropped
+			}, 0)
+	}
+
+	/**
+	 * If some dropped procs should not be considered as bad, override this with logic to return false for drops that are not considered bad.
+	 * Recommended to have your override return super.considerDroppedProcs && your logic, to still get the death and downtime forgiveness without having to copy the logic
+	 * @param window The window that included dropped procs
+	 * @returns True if the drop should count as bad, False if the drop should not be counted
+	 */
+	protected considerDroppedProcs(window: ProcBuffWindow): boolean {
+		// Don't count procs that were dropped due to death
+		if (this.actors.get(this.parser.actor).at(window.stop).hp.current === 0) { return false }
+
+		// Don't count procs that were dropped due to downtime
+		if (this.downtime.isDowntime(window.stop)) { return false }
+
+		// Don't count procs that were dropped due to end of fight
+		if (window.stop === this.parser.pull.timestamp + this.parser.pull.duration) { return false }
```

Comment:
just in case there's ever funky numbers
```suggestion
		if (window.stop >= this.parser.pull.timestamp + this.parser.pull.duration) { return false }
```